### PR TITLE
added node built in apis to work in react view

### PR DESCRIPTION
**Pull request by gustavoFreireS**

### CODE REVIEW

File: electron/electron.js [hunk 1]
```diff
@@ -10,14 +9,31 @@ let mainWindow;
...
+      nodeIntegration: true,
+      nativeWindowOpen: true,
+    },
+    width: 300,
+    height: 130,
     show: false,
     titleBarStyle: 'hiddenInset',
     alwaysOnTop: true,
   });
+  mainWindow.webContents.on('new-window', (event, url, frameName, disposition, options, additionalFeatures) => {
```
Comment (gustavoFreireS) [inline electron/electron.js]: this block here ensures a new window with this configuration can be opened (for use with terminal in the future, or any other thing)

**Comment by davidcostadev** [comment]
@gustavoFreireS fix the conflicts

**Comment by davidcostadev** [comment]
First fix lint:
```
/Users/davidcosta/workplace/electron-dojo-timer/src/containers/HomePage/TestConfig.js
   6:9   error  Use object destructuring                                                                                                                                            prefer-destructuring
   6:18  error  Unexpected require()                                                                                                                                                global-require
   6:18  error  'electron' should be listed in the project's dependencies, not devDependencies                                                                                      import/no-extraneous-dependencies
  11:31  error  Insert `,`                                                                                                                                                          prettier/prettier
  12:7   error  Insert `;`                                                                                                                                                          prettier/prettier
  13:5   error  Unexpected console statement                                                                                                                                        no-console
  16:11  error  'remote' is already declared in the upper scope                                                                                                                     no-shadow
  16:11  error  Use object destructuring                                                                                                                                            prefer-destructuring
  16:20  error  'electron' should be listed in the project's dependencies, not devDependencies                                                                                      import/no-extraneous-dependencies
  16:20  error  Unexpected require()                                                                                                                                                global-require
  17:11  error  'remoteObj' is already declared in the upper scope                                                                                                                  no-shadow
  17:23  error  Delete `·`                                                                                                                                                          prettier/prettier
  20:4   error  Insert `;`                                                                                                                                                          prettier/prettier
  21:18  error  Replace `(e)·=>` with `e·=>·`                                                                                                                                       prettier/prettier
  23:5   error  Unexpected console statement                                                                                                                                        no-console
  26:5   error  Unexpected console statement                                                                                                                                        no-console
  27:4   error  Insert `;`                                                                                                                                                          prettier/prettier
  33:6   error  Insert `,`                                                                                                                                                          prettier/prettier
  34:4   error  Insert `;`                                                                                                                                                          prettier/prettier
  38:79  error  A space is required before closing bracket                                                                                                                          react/jsx-tag-spacing
  38:79  error  Insert `·`                                                                                                                                                          prettier/prettier
  40:17  error  Replace `·="command"·defaultValue={config.command}·id="command"·onChange={change}` with `="command"·defaultValue={config.command}·id="command"·onChange={change}·`  prettier/prettier
  40:18  error  There should be no space before '='                                                                                                                                 react/jsx-equals-spacing
  40:89  error  A space is required before closing bracket                                                                                                                          react/jsx-tag-spacing
  41:7   error  Missing an explicit type attribute for button                                                                                                                       react/button-has-type
  41:35  error  Expected only one space between "style" and "onClick"                                                                                                               react/jsx-props-no-multi-spaces
  41:35  error  Replace `onClick={save}>salvar` with `·onClick={save}>⏎········salvar⏎······`                                                                                       prettier/prettier
  46:27  error  Insert `⏎`                                                                                                                                                          prettier/prettier

/Users/davidcosta/workplace/electron-dojo-timer/src/containers/HomePage/Timer.js
    2:22  error  Replace `"react-dom"` with `'react-dom'`                                                                                                                                                                       prettier/prettier
    5:1   error  `./TestConfig` import should occur after import of `xterm`                                                                                                                                                     import/order
    6:22  error  Unable to resolve path to module 'xterm/lib/addons/fit/fit'                                                                                                                                                    import/no-unresolved
    7:22  error  Unable to resolve path to module 'node-pty'                                                                                                                                                                    import/no-unresolved
    8:1   error  'os' should be listed in the project's dependencies. Run 'npm i -S os' to add it                                                                                                                               import/no-extraneous-dependencies
    9:26  error  Unable to resolve path to module 'xterm'                                                                                                                                                                       import/no-unresolved
   51:9   error  'modal' is never reassigned. Use 'const' instead                                                                                                                                                               prefer-const
   53:67  error  Replace `"modal"` with `'modal'`                                                                                                                                                                               prettier/prettier
   54:4   error  Insert `;`                                                                                                                                                                                                     prettier/prettier
   56:11  error  Use object destructuring                                                                                                                                                                                       prefer-destructuring
   56:18  error  Unexpected require()                                                                                                                                                                                           global-require
   57:11  error  Use object destructuring                                                                                                                                                                                       prefer-destructuring
   57:20  error  'electron' should be listed in the project's dependencies, not devDependencies                                                                                                                                 import/no-extraneous-dependencies
   57:20  error  Unexpected require()                                                                                                                                                                                           global-require
   63:39  error  Insert `,`                                                                                                                                                                                                     prettier/prettier
   65:9   error  'terminal' is never reassigned. Use 'const' instead                                                                                                                                                            prefer-const
   83:8   error  Insert `,`                                                                                                                                                                                                     prettier/prettier
   87:31  error  Replace `os.platform()·===·'win32'·?·'powershell.exe'·:·process.env.SHELL·||·'/bin/bash',·[],` with `⏎······os.platform()·===·'win32'·?·'powershell.exe'·:·process.env.SHELL·||·'/bin/bash',⏎······[],⏎·····`  prettier/prettier
   88:7   error  Insert `··`                                                                                                                                                                                                    prettier/prettier
   89:7   error  Replace `rows:·term.rows` with `··rows:·term.rows,`                                                                                                                                                            prettier/prettier
   90:1   error  Replace `····}` with `······},⏎····`                                                                                                                                                                           prettier/prettier
   91:21  error  Replace `(data)` with `data`                                                                                                                                                                                   prettier/prettier
  100:50  error  Insert `,`                                                                                                                                                                                                     prettier/prettier
  105:7   error  Unexpected console statement                                                                                                                                                                                   no-console
  106:7   error  Unexpected console statement                                                                                                                                                                                   no-console
  107:7   error  Unexpected console statement                                                                                                                                                                                   no-console
  111:8   error  Delete `⏎·····`                                                                                                                                                                                                prettier/prettier
  114:8   error  Delete `⏎`                                                                                                                                                                                                     prettier/prettier
  117:5   error  Unexpected console statement                                                                                                                                                                                   no-console
  118:4   error  Insert `;`                                                                                                                                                                                                     prettier/prettier
  138:94  error  Replace `<i·className="fa·fa-stop"·/>` with `⏎············<i·className="fa·fa-stop"·/>⏎··········`                                                                                                             prettier/prettier
  140:9   error  Missing an explicit type attribute for button                                                                                                                                                                  react/button-has-type
  140:78  error  Replace `<i·className="fa·fa-bolt"·/>` with `⏎··········<i·className="fa·fa-bolt"·/>⏎········`                                                                                                                 prettier/prettier
  141:9   error  Missing an explicit type attribute for button                                                                                                                                                                  react/button-has-type
  141:80  error  Replace `<i·className="fa·fa-cog"·/>` with `⏎··········<i·className="fa·fa-cog"·/>⏎········`                                                                                                                   prettier/prettier

✖ 63 problems (63 errors, 0 warnings)
```

File: src/containers/HomePage/Timer.js [hunk 2]
```diff
@@ -40,9 +47,78 @@ const Timer = () => {
     setCurrentState('ended');
     setCounter(THRESHOLD);
   }
+  const config = () => {
+    let modal = window.open('', 'modal');
+    modal.document.write('<div id="modal"/>');
+    ReactDOM.render(<TestConfig />, modal.document.getElementById("modal"));
+  }
+  const test = () => {
```
Comment (davidcostadev) [inline src/containers/HomePage/Timer.js]: Please, could you remove this code from here and put this in another file.

File: src/containers/HomePage/Timer.js [hunk 3]
```diff
@@ -40,9 +47,78 @@ const Timer = () => {
...
+
+      if (!error) {
+        setColor('#4daf7c');
+      }
+      else {
+        setColor('red');
+      }
+
+    });
+    console.log(remoteObj.testPath);
```
Comment (davidcostadev) [inline src/containers/HomePage/Timer.js]: Remove this comment

File: src/containers/HomePage/TestConfig.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+const TestConfig = () => {
+  const [config, setConfig] = useState({});
+  const remote = require('electron').remote;
+  const remoteObj = remote.getGlobal('sharedObject');
+  useEffect(() => {
+    setConfig({
+      command: remoteObj.testCommand,
+      path: remoteObj.testPath
+    })
+    console.log(config);
```
Comment (davidcostadev) [inline src/containers/HomePage/TestConfig.js]: remove this console.log

File: src/containers/HomePage/TestConfig.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    const remoteObj =  remote.getGlobal('sharedObject');
+    remoteObj.testCommand = config.command;
+    remoteObj.testPath = config.path;
+  }
+  const change = (e) =>{
+    const newConfig = config;
+    console.log(e.target.id);
+    newConfig[e.target.id] = e.target.value;
+    setConfig(newConfig);
+    console.log(config);
```
Comment (davidcostadev) [inline src/containers/HomePage/TestConfig.js]: remove this console.log

**Comment by davidcostadev** [comment]
Please add some gif too

**Pull request merged**